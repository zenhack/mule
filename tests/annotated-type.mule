#XFAIL: expand_path
(fn x. x.f) : (y : { type t, f : t }) -> y.t
