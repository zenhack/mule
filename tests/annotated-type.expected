
Desugared:

	((: ((y : (exist $0 (record (types ((t $0))) (values ((f t)))))) -> (. y t)))
	 (fn x ((. f) x)))

inferred type:

	(all a ((record (types ((t a))) (values ((f a)))) -> a))

Runtime term:

	(Lam (0 () (App ((GetField f) (Var 0)))))

Evaluated:

	(Lam (0 () (App ((GetField f) (Var 0)))))
